### DISCUSSION TITLE
[2022/03/29] Frontend Night

### DISCUSSION THREAD

**Discussion by jjenko** [wantedly/frontend-night#198] [Archive]
## Policy

- 資料の準備をがんばらない
- 参加・途中退席は自由
- スキップしない

話したい・聞きたいネタを書いてく ✏️

ハッシュタグ [#frontend_night](https://twitter.com/hashtag/frontend_night?src=hash)

Prev: https://github.com/wantedly/frontend-night/discussions/197 

 _Created from https://github.com/wantedly/frontend-night/discussions/179 by [issue-creator](https://github.com/rerost/issue-creator)_

**Comment by kobayang** [comment]
## [Floating UI](https://floating-ui.com/)

- ReactDOM https://floating-ui.com/docs/react-dom-interactions
- hooksベースで primitive な floating UI (e.g. Tooltip) を作ることができるライブラリ
  - props のI/Fなど、Tooltip 作る時の参考にできそう
    - かなり react-popper ににている
- react-popper を使っているけど React v17以降との噛み合わせが悪いので代替として検討しても良いかも

**Reply by kobayang** [reply]
- popper(v2) → floating UI に変わった。popper 自体は Pure JS なライブラリだった。
  - それが faloting UI になることで React 前提になった。
  - `@floating-ui/core` が元々の popper？
- react-popper そもそも popper の wrapper だったので、floating UI に置き換えても良い気がする。
  - まだ minor version なのでそこは注意が必要。

**Reply by kobayang** [reply]
https://floating-ui.com/docs/motivation
色々良くなってそう

**Reply by kobayang** [reply]
react-popper 自体は popper v1 に依存している

**Comment by kobayang** [comment]
## [デザインシステムのa11y対応に役立つ新しいWeb Standard](https://speakerdeck.com/line_developers/useful-new-web-standards-for-a11y-in-design-systems)

- デザインシステムとa11yの話
  - ちゃんとしたデザインシステムはa11yの土台にはなるが保証はしない
    - 各ページ単位で担おうね、という話だと理解したが違うかもしれない
- テクニックとして以下が紹介されてた
  - `<dialog>`
  - motion 減らす CSS media feature
  - focus-visible

**Reply by kobayang** [reply]
Safari 15.4 だけど、実際使っていいの？という話

> よって、iOSのシェア率の90%以上は常に2年以内にリリースした最新の3つのメジャーバージョンとなっています。すなわちSafariのバージョンも2年以内にリリースされたものに対応する必要があります。
https://engineering.linecorp.com/ja/blog/uit-baseline-for-front-end-development-in-2022/

によれば 2年ぐらいでSafariのversionの切り替えが90%ほど完了するから、それぐらい待つ必要がありそう？

**Comment by kobayang** [comment]
## [WAI-ARIA オーサリング・プラクティス](https://waic.jp/docs/2019/NOTE-wai-aria-practices-1.1-20190207/#intro)

- [WCAG2.1](https://waic.jp/docs/WCAG21/) でどんなa11yを満たすべきかの基準はあるが、それぞれのUIにどう適応できるかの具体化が難しい（と個人的に思ってた）
- [デザイン・パターンとウィジェット](https://waic.jp/docs/2019/NOTE-wai-aria-practices-1.1-20190207/#aria_ex) はそれぞれのデザインパターン(≒UI）が満たすべきキーボード・インタラクションとWAI-ARIAを紹介している。
  - 特にキーボード・インタラクションのセクションが嬉しい。
  - 汎用的なコンポーネント(e.g. デザインシステムライブラリ）を作る上で役立ちそう。
- 例: [ダイアログ（モーダル）](https://waic.jp/docs/2019/NOTE-wai-aria-practices-1.1-20190207/#dialog_modal)

全部理解する必要はなくて、こういうリストがあることをインデックスしておくと良さそう。

**Reply by kobayang** [reply]
react-aria や headless ui ライブラリのコードを読む際にも理解の役に立つはず

**Reply by kobayang** [reply]
ariakit: react-aria  みたいなやつ（良さそう）
https://github.com/ariakit/ariakit

**Comment by kobayang** [comment]
## ES2022 RC

- [ES2022 のRCがリリースされた](https://github.com/tc39/ecma262/releases/tag/es2022-candidate-2022-03)。正式リリースは6月。
- [ES2022で追加される機能一覧](https://gist.github.com/azu/a33211926e3de8bff8de83d6783c55eb)

**Reply by kobayang** [reply]
- 一覧にある Annex B とは → [JavaScriptのレガシー挙動を定めたAnnex Bをひたすら読む記事](https://zenn.dev/qnighy/articles/1d96f2c0c662f6)

**Comment by kobayang** [comment]
## CSS Tips

- [よくあるWeb課題をCSSで乗り切る！知っておきたい最新テクニック20選](https://photoshopvip.net/134168)
- [よくあるWeb制作の悩みをCSSで解決！最新テクニック40選まとめ](https://photoshopvip.net/131693)

ちゃんと最新テクニックだった

**Reply by kobayang** [reply]
selectmenu https://css-tricks.com/the-selectmenu-element/。今はChromeだけ対応。

（selectbox は実装がとても難しいから、Web標準に入ってくれるととても嬉しい）
